Tests/MongoSwiftTests/AsyncAwaitTests.swift

```diff
@@ -0,0 +1,90 @@
+#if compiler(>=5.5) && canImport(_Concurrency) && os(Linux)
+
+import Foundation
+@testable import MongoSwift
+import Nimble
+import NIO
+import TestsCommon
+import XCTest
+
+/// Temporary utility function until XCTest supports `async` tests.
+func testAsync(_ block: @escaping () async throws -> Void) {
+    let group = DispatchGroup()
+    group.enter()
+    Task.detached {
+        try await block()
+        group.leave()
+    }
+    group.wait()
+}
+
+/// Asserts that the provided block returns true within the specified timeout. Nimble's `toEventually` can only be used
+/// rom the main testing thread which is too restrictive for our purposes testing the async/await APIs.
+func assertIsEventuallyTrue(_ block: () -> Bool, description: String, timeout: TimeInterval = 5) {
+    let start = DispatchTime.now()
+    while DispatchTime.now() < start + timeout {
+        if block() {
+            return
```

Comment:
yeah good point, I added a configurable `sleepInterval` to this method; this required making the method `async` and `throws`. 

I also reordered the parameters because I wanted to be able to pass trailing closures 🙂